op-acceptance-tests/tests/flashblocks/flashblocks_stream_test.go

```diff
@@ -51,38 +59,103 @@ func TestFlashblocksStream(gt *testing.T) {
+	expectedChainID := sys.L2Chain.ChainID().ToBig()
+	require.Equal(t, oprbuilderNode.Escape().ChainID().ToBig(), expectedChainID, "flashblocks builder node chain id should match expected chain id")
+
+	driveViaTestSequencer(t, sys, 3)
+
+	// testFlashblocksStreamRbuilder(t, logger, oprbuilderNode, FlashblocksStreamMode_Leader, flashblocksStreamRateMs)
+
+	// Test the presence / absence of a flashblocks stream operating at a 250ms (configurable via env var FLASHBLOCKS_STREAM_RATE) rate from a flashblocks-websocket-proxy node.
+	// Allow a generous window for first flashblocks to appear.
+	testDuration := time.Duration(int64(flashblocksStreamRateMs*maxExpectedFlashblocks*2)) * time.Millisecond
+	failureTolerance := int(0.15 * float64(maxExpectedFlashblocks))
+
+	logger.Debug("Test duration", "duration", testDuration, "failure tolerance (of flashblocks)", failureTolerance)
+
+	require.NotNil(t, rollupBoostNode.Escape(), "rollupBoostNode should not be nil")
+
+	// Instrument builder stream separately to confirm flashblocks emission upstream.
+	builderOutput := make(chan []byte, maxExpectedFlashblocks)
+	builderDone := make(chan struct{})
+	go func() {
+		err := oprbuilderNode.FlashblocksClient().ListenFor(logger.With("stream_source", "op-rbuilder"), testDuration, builderOutput, builderDone)
+		require.NoError(t, err)
+	}()
+	builderMessages := make([]string, 0)
+
+	output := make(chan []byte, maxExpectedFlashblocks)
+	doneListening := make(chan struct{})
+	streamedMessages := make([]string, 0)
+	go func() {
+		err := rollupBoostNode.FlashblocksClient().ListenFor(logger.With("stream_source", "rollup-boost"), testDuration, output, doneListening)
+		require.NoError(t, err)
+	}()
+
+	listening := true
+	for listening {
+		select {
+		case <-doneListening:
+			doneListening = nil
+		case <-builderDone:
+			builderDone = nil
+		case msg := <-output:
+			streamedMessages = append(streamedMessages, string(msg))
+		case msg := <-builderOutput:
+			builderMessages = append(builderMessages, string(msg))
+		}
 
-				mode := FlashblocksStreamMode_Follower
-				if dsl.NewConductor(flashblocksBuilderNode.Escape().Conductor()).IsLeader() {
-					mode = FlashblocksStreamMode_Leader
-				}
+		if doneListening == nil && builderDone == nil {
+			listening = false
+		}
+	}
 
-				testFlashblocksStreamRbuilder(tt, logger, flashblocksBuilderNode, mode, flashblocksStreamRateMs)
-			}
+	defer close(output)
+	defer close(builderOutput)
 
-			for _, flashblocksWebsocketProxy := range sys.FlashblocksWebsocketProxies[l2Chain] {
-				testFlashblocksStreamFbWsProxy(tt, logger, flashblocksWebsocketProxy, flashblocksStreamRateMs)
-			}
-		})
+	logger.Info("Completed WebSocket stream reading", "message_count", len(streamedMessages))
+	logger.Info("Completed builder WebSocket stream reading", "message_count", len(builderMessages))
```

Comment:
nit: might be nice to combine the logs

```suggestion
	logger.Info("Completed WebSocket stream reading", "msg_count", len(streamedMessages), "builder_msg_count", len(builderMessages))
```